### PR TITLE
Drive recent sessions display with field set

### PR DESCRIPTION
**Pull request by jjbennett**
# Critical Changes

# Changes
- The Recent Sessions Home Page component output is now driven by a field set.

# Issues Closed

# New Metadata

# Deleted Metadata

# Definition of Done
  Refer to [Asteroids DoD document](https://salesforce.quip.com/iq2mAy4i62oM) to see any additional details for the items below
~~Place holder data is incorporated (Refer to [Case Management Sample Data document](https://quip.com/cbFcAPJF8t0z))~~
- [x] Base axe-core JEST test included for any new LWC (No critical violations)
- [x] CRUD/FLS is enforced in Apex
- [x] Permission sets & field sets are updated to account for CRUD/FLS/TABS for new object metadata
- [x] All modified classes are unit tested (Apex) at 100% coverage
- [x] UX approval or UX not necessary
- [x] PR contains draft release notes
- [x] Attach work item to the pull request with [Lurch](https://salesforce.quip.com/50ZRA5LEWVzH) `**Lurch:attach W-000000` or `**Lurch:add`
- [ ] All **acceptance criteria** have been met
    - [ ] Developer
    - [ ] Code Reviewer
    - [ ] QA
- [ ] QE story level testing completed

### CODE REVIEW

**Comment by jjbennett** [comment]
[W-8091296](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008IsNeIAK/view)

File: force-app/main/default/lwc/formattedField/__tests__/formattedField.test.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+        return global.flushPromises().then(async () => {
+            const lightningFormattedText = element.shadowRoot.querySelector(
+                "lightning-formatted-text"
+            );
+            expect(lightningFormattedText.value).toBe(mockRecord.Name);
+            await expect(element).toBeAccessible();
+        });
+    });
+    it("displays a lightning formatted text field when a relatd value is present", async () => {
```
Comment (bethbrains) [inline force-app/main/default/lwc/formattedField/__tests__/formattedField.test.js:52]: typo

File: force-app/main/default/lwc/formattedField/__tests__/formattedField.test.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+        return global.flushPromises().then(async () => {
+            const lightningFormattedText = element.shadowRoot.querySelector(
+                "lightning-formatted-text"
+            );
+            expect(lightningFormattedText.value).toBe(mockRecord.Account.Name);
+            await expect(element).toBeAccessible();
+        });
+    });
+    it("displays a lightning formatted text field when a relatd value is not present", async () => {
```
Comment (bethbrains) [inline force-app/main/default/lwc/formattedField/__tests__/formattedField.test.js:67]: typo

File: force-app/main/default/lwc/formattedField/__tests__/formattedField.test.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+        return global.flushPromises().then(async () => {
+            const lightningFormattedText = element.shadowRoot.querySelector(
+                "lightning-formatted-text"
+            );
+            expect(lightningFormattedText.value).toBeUndefined();
+            await expect(element).toBeAccessible();
+        });
+    });
+    it("dispalys a lightning formatted text field when a value is missing", async () => {
```
Comment (bethbrains) [inline force-app/main/default/lwc/formattedField/__tests__/formattedField.test.js:81]: typo

File: force-app/main/default/lwc/formattedField/__tests__/formattedField.test.js [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        return global.flushPromises().then(async () => {
+            const lightningFormattedText = element.shadowRoot.querySelector(
+                "lightning-formatted-text"
+            );
+            expect(lightningFormattedText.value).toBeUndefined();
+            await expect(element).toBeAccessible();
+        });
+    });
+
+    it("dispalys a lightning formatted text field when the record is missing", async () => {
```
Comment (bethbrains) [inline force-app/main/default/lwc/formattedField/__tests__/formattedField.test.js:97]: typo

File: force-app/main/default/lwc/formattedField/__tests__/formattedField.test.js [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+        return global.flushPromises().then(async () => {
+            const lightningFormattedText = element.shadowRoot.querySelector(
+                "lightning-formatted-text"
+            );
+            expect(lightningFormattedText.value).toBeUndefined();
+            await expect(element).toBeAccessible();
+        });
+    });
+
+    it("dispalys a lightning formatted date time field when type is date", async () => {
```
Comment (bethbrains) [inline force-app/main/default/lwc/formattedField/__tests__/formattedField.test.js:110]: typo

File: force-app/main/default/lwc/formattedField/__tests__/formattedField.test.js [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+        return global.flushPromises().then(async () => {
+            const lightningFormattedDateTime = element.shadowRoot.querySelector(
+                "lightning-formatted-date-time"
+            );
+            expect(lightningFormattedDateTime.value).toBe(today);
+            await expect(element).toBeAccessible();
+        });
+    });
+
+    it("dispalys a lightning formatted date time field when type is date-time", async () => {
```
Comment (bethbrains) [inline force-app/main/default/lwc/formattedField/__tests__/formattedField.test.js:129]: typo

File: force-app/main/default/lwc/formattedField/__tests__/formattedField.test.js [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+        return global.flushPromises().then(async () => {
+            const lightningFormattedDateTime = element.shadowRoot.querySelector(
+                "lightning-formatted-date-time"
+            );
+            expect(lightningFormattedDateTime.value).toBe(today);
+            await expect(element).toBeAccessible();
+        });
+    });
+
+    it("dispalys a lightning formatted time field when type is time", async () => {
```
Comment (bethbrains) [inline force-app/main/default/lwc/formattedField/__tests__/formattedField.test.js:148]: typo

File: force-app/main/default/permissionsets/PMDM_View.permissionset-meta.xml [hunk 8]
```diff
@@ -4,6 +4,18 @@
...
+        <enabled>true</enabled>
+    </classAccesses>
+    <classAccesses>
+        <apexClass>FieldSetController</apexClass>
+        <enabled>true</enabled>
+    </classAccesses>
+    <classAccesses>
+        <apexClass>RecentServiceSessionController</apexClass>
+        <enabled>true</enabled>
+    </classAccesses>
```
Comment (bethbrains) [inline force-app/main/default/permissionsets/PMDM_View.permissionset-meta.xml:18]: these are already on the other perm sets?

File: force-app/main/default/lwc/sessionCard/sessionCard.html [hunk 9]
```diff
@@ -35,33 +36,31 @@
...
+                        <lightning-layout-item key={it.value.apiName}>
+                            <c-formatted-field
+                                field={it.value}
+                                record={session}
+                            ></c-formatted-field>
+                            <template if:false={it.last}
+                                ><span
+                                    class="slds-var-p-left_xx-small slds-var-p-right_xx-small"
+                                    >{separator}</span
+                                ></template
```
Comment (bethbrains) [inline force-app/main/default/lwc/sessionCard/sessionCard.html:52]: super interesting. i didn't know this was an option. thanks!! fyi @bmuniswamy

**Review by bethbrains — APPROVED** [review]
some typos but nothing blocking QE. looks really good!! clever stuff, @jjbennett !

File: force-app/main/default/permissionsets/PMDM_View.permissionset-meta.xml (hunk 8 above)
Comment (jjbennett) [reply force-app/main/default/permissionsets/PMDM_View.permissionset-meta.xml:18]: Indeed!

**Review by lbeller — APPROVED** [review]

**Pull request merged**